inference/server/oasst_inference_server/routes/chats.py

```diff
@@ -133,19 +134,33 @@ async def message_events(
     if message.has_finished:
         raise fastapi.HTTPException(status_code=204, detail=message.state)
 
-    async def event_generator(chat_id: str, message_id: str):
+    async def event_generator(chat_id: str, message_id: str, worker_compat_hash: str | None):
         redis_client = deps.make_redis_client()
-        queue = queueing.message_queue(redis_client, message_id=message_id)
+        message_queue = queueing.message_queue(redis_client, message_id=message_id)
+        work_queue = (
+            queueing.work_queue(redis_client, worker_compat_hash=worker_compat_hash)
+            if worker_compat_hash is not None
+            else None
+        )
         has_started = False
         try:
             while True:
-                item = await queue.dequeue(timeout=settings.pending_event_interval)
+                item = await message_queue.dequeue(timeout=settings.pending_event_interval)
                 if item is None:
                     if not has_started:
+                        if work_queue is None:
+                            qpos, qlen = 0, 1
+                        else:
+                            # TODO: make more efficient, e.g. pipeline
+                            qdeq = await work_queue.get_deq_counter()
+                            qenq = await work_queue.get_enq_counter()
+                            mpos = await queueing.get_pos_value(redis_client, message_id)
```

Comment:
could this be done in parallel like this? or is this a race condition?

```python
[qdeq, qenq, mpos] = await asyncio.gather(
  work_queue.get_deq_counter(),
  work_queue.get_enq_counter(),
  queueing.get_pos_value(redis_client, message_id)
)
```

Reply:
good idea